### PR TITLE
rgw: add an option to clear all usage entries

### PR DESCRIPTION
**Pull request by theanalyst**
This is built atop of https://github.com/ceph/ceph/pull/19131 (though
not dependant exactly, tests will be similar so will be rebased once
that is merged), adding an option to clear all the usage logs by using
the omap clear cls api, which would be faster for clearing usage log
entries of all users, we can't yet do this on a single user because
multiple users may map to a single oid. Adds a radosgw-admin cli command
called usage clear and implements these in the cls_rgw and rados layer

### CODE REVIEW

File: src/cls/rgw/cls_rgw.cc [hunk 1]
```diff
@@ -3078,9 +3074,26 @@ int rgw_user_usage_log_trim(cls_method_context_t hctx, bufferlist *in, bufferlis
...
+  if (!more && iter.empty())
+    return -ENODATA;
+
   return 0;
 }
 
+int rgw_usage_log_clear(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
+{
+  CLS_LOG(10,"%s", __func__);
+  /* only continue if object exists! */
```
Comment (cbodley) [inline src/cls/rgw/cls_rgw.cc]: why is this stat needed? will map_clear not do the right thing if the object doesn't exist?

File: src/test/cls_rgw/test_cls_rgw.cc [hunk 2]
```diff
@@ -629,6 +629,47 @@ TEST(cls_rgw, gc_defer)
...
+  // delete and read to assert that we've deleted all the values
+  ASSERT_EQ(0, cls_rgw_usage_log_trim(ioctx, oid, user, start_epoch, end_epoch));
+
+
+  ret = cls_rgw_usage_log_read(ioctx, oid, user, start_epoch, end_epoch,
+			       max_entries, read_iter, usage2, &truncated);
+  ASSERT_EQ(0, ret);
+  ASSERT_EQ(0, usage2.size());
+
+}
```
Comment (cbodley) [inline src/test/cls_rgw/test_cls_rgw.cc]: could you please add some coverage for usage_log_clear() as well? it sounds like both the exist and noexist cases are interesting

**Comment by theanalyst** [comment]
@cbodley rebased and added new tests

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-02-14-2124](https://trello.com/c/guwy7LoB/261-wip-yuri3-testing-2018-02-14-2124)

**Pull request merged**